File: go-version/leetcode/LC781/num_rabbits.go
package LC781

/**
 * @Author: cutejiuge cutejiuge@163.com
 * @Date: 2025/4/20 上午8:09
 * @FilePath: num_rabbits
 * @Description: 分组计数
 */

func numRabbits(answers []int) int {
	var ans int
	mp := make(map[int]int)
	for _, num := range answers {
		if mp[num] == 0 {
			ans += num + 1
			mp[num] = num
		} else {
			mp[num]--
		}
	}
	return ans
}

File: python-version/LC781/num_rabbits.py
from collections import defaultdict


class Solution:
    @staticmethod
    def numRabbits(answers: list[int]) -> int:
        ans = 0
        mp = defaultdict(int)
        for a in answers:
            if mp[a] == 0:
                ans += a + 1
                mp[a] = a
            else:
                mp[a] -= 1
        return ans
